src/lambdas/users-create.js

```diff
@@ -0,0 +1,107 @@
+import ExceptionHandler from '@/exceptions/handler';
+import User from '@/models/user';
+import {DataMapper} from '@aws/dynamodb-data-mapper';
+
+const DyanamoDB = require('aws-sdk/clients/dynamodb');
+const Lambda = require('aws-sdk/clients/lambda');
+
+/**
+ * @apiDefine Headers
+ *
+ * @apiHeader {String} [x-api-key] ApiGateway API key. Required if stack was created with `ApiKeyRequired` set to "Yes".
+ * @apiHeaderExample {json} Header-Example
+ * {
+ * 	"x-api-key": "API_KEY_HERE"
+ * }
+ */
+
+/**
+ * @api {post} /users Create a User
+ * @apiGroup Users
+ *
+ * @apiExample {curl} Example
+ * curl -X POST {{API_URL}}/users \
+ * -H 'x-api-key: API_KEY_HERE' \
+ * -d '{"body":{"name":"Jane Doe","dob":"1980-04-08","address": {"street": "140 Madison St","city": "Rosemount","state": "MN","zip": "55068"},"description": "Likes waffles."}}'
+ *
+ * @apiSampleRequest off
+ *
+ * @apiUse Headers
+ *
+ * @apiParam (Body) {String} name User's name.
+ * @apiParam (Body) {String} [dob] User's date of birth.
+ * @apiParam (Body) {Object} [address] User's address.
+ * @apiParam (Body) {String} [description] Description of the User.
+ *
+ * @apiParamExample {json} Request-Example
+ * {
+ * 	"name": "Jennie Smith",
+ * 	"dob": "1990-05-08",
+ * 	"address": {
+ * 		"street": "1990-05-08",
+ * 		"city": "Moncks Corner",
+ * 		"state": "SC",
+ * 		"zip": "29461"
+ * 	},
+ * 	"description": "She's like the best."
+ * }
+ *
+ * @apiSuccess (Success) {String} id User's unique identifier.
+ * @apiSuccess (Success) {String} name User's name.
+ * @apiSuccess (Success) {String} dob ISO 8601 Date of User's date of birth.
+ * @apiSuccess (Success) {Object} address User's address.
+ * @apiSuccess (Success) {String} description Description of the User.
+ * @apiSuccess (Success) {String} createdAt ISO 8601 Datetime in UTC of when the User was created.
+ * @apiSuccess (Success) {String} updatedAt ISO 8601 Datetime in UTC of when the User was last updated.
+ *
+ * @apiSuccessExample Success-Response
+ * HTTP/1.1 201 Created
+ * {
+ * 	"data":
+ * 		"id": "e34c2498-8b12-485a-9f3b-92342b561062",
+ * 		"name": "Jennie Smith",
+ * 		"dob": "1990-05-08",
+ * 		"address": {
+ * 			"street": "939 Elmwood Street",
+ * 			"city": "Moncks Corner",
+ * 			"state": "SC",
+ * 			"zip": "29461"
+ * 		},
+ * 		"description": "She's like the best.",
+ * 		"createdAt": "2019-10-21T02:23:31.467Z",
+ * 		"updatedAt": "2019-10-21T02:23:31.467Z"
+ * 	}
+ * }
+ *
+ * @apiUse InvalidAttributeException
+ * @apiUse Exception
+ */
+export async function handler(event, context, callback) {
+	console.log(event);
+
+	try {
+		let user = new User(event.body);
+		await user.validate();
+		await user.address.validate();
+
+		const client = new DyanamoDB();
+		const mapper = new DataMapper({client});
+
+		user = await mapper.put(user);
+
+		// Clear the CloudFront cache
+		const lambda = new Lambda();
+		const params = {
```

Comment:
The reason that this lives in a separate Lambda function in the first place is because passing the `DISTRIBUTION_ID` environment variable to the API functions caused a circular dependency. There are several ways to handle this, the ones I was deciding between where:
1. Invoke a separate function that could live outside the circular dependency.
2. Load the distribution id from the Systems Manager Parameter Store.

I also debated whether or not throw the error if cache invalidation fails, and thinking through it again, it should probably just log an move on. Update incoming.